diff --git a/grails-aether/src/main/groovy/org/codehaus/groovy/grails/resolve/maven/aether/AetherDependencyManager.groovy b/grails-aether/src/main/groovy/org/codehaus/groovy/grails/resolve/maven/aether/AetherDependencyManager.groovy
index d73e3b5..51b144a 100644
--- a/grails-aether/src/main/groovy/org/codehaus/groovy/grails/resolve/maven/aether/AetherDependencyManager.groovy
+++ b/grails-aether/src/main/groovy/org/codehaus/groovy/grails/resolve/maven/aether/AetherDependencyManager.groovy
@@ -522,7 +522,7 @@ class AetherDependencyManager implements DependencyManager {
         grailsDependenciesByScope[dependency.scope] << grailsDependency
         final aetherDependencies = dependencies
         aetherDependencies << dependency
-        if (dependency.artifact.groupId == 'org.grails.plugins' || dependency.artifact.properties.extension == 'zip') {
+        if (dependency.artifact.groupId == 'org.grails.plugins' || dependency.artifact.extension == 'zip') {
             grailsPluginDependencies << grailsDependency
             grailsPluginDependenciesByScope[dependency.scope] << grailsDependency
         }
@@ -531,6 +531,7 @@ class AetherDependencyManager implements DependencyManager {
     protected org.codehaus.groovy.grails.resolve.Dependency createGrailsDependency(Dependency dependency, DependencyConfiguration configuration = null) {
         Artifact artifact = dependency.artifact
         final grailsDependency = new org.codehaus.groovy.grails.resolve.Dependency(artifact.groupId, artifact.artifactId, artifact.version)
+        grailsDependency.extension = artifact.extension
         if (configuration) {
             grailsDependency.transitive = configuration.transitive
             grailsDependency.exported = configuration.exported
@@ -564,7 +565,7 @@ class AetherDependencyManager implements DependencyManager {
             grailsDependencies << dependency
             grailsDependenciesByScope["build"] << dependency
             buildDependencies << mavenDependency
-            if (dependency.group == 'org.grails.plugins' || dependency.properties.extension == 'zip') {
+            if (dependency.group == 'org.grails.plugins' || dependency.extension == 'zip') {
                 grailsPluginDependencies << dependency
                 grailsPluginDependenciesByScope["build"] << dependency
             }
@@ -602,7 +603,7 @@ class AetherDependencyManager implements DependencyManager {
     }
 
     protected boolean isGrailsPlugin(org.codehaus.groovy.grails.resolve.Dependency dependency) {
-        dependency.group == 'org.grails.plugins' || dependency.properties.extension == 'zip'
+        dependency.group == 'org.grails.plugins' || dependency.extension == 'zip'
     }
 
     @Override
diff --git a/grails-bootstrap/src/main/groovy/org/codehaus/groovy/grails/resolve/Dependency.groovy b/grails-bootstrap/src/main/groovy/org/codehaus/groovy/grails/resolve/Dependency.groovy
index 01ad729..86e6149 100644
--- a/grails-bootstrap/src/main/groovy/org/codehaus/groovy/grails/resolve/Dependency.groovy
+++ b/grails-bootstrap/src/main/groovy/org/codehaus/groovy/grails/resolve/Dependency.groovy
@@ -36,6 +36,7 @@ class Dependency {
     boolean transitive = true
     boolean inherited = false
     boolean exported = true
+    String extension = null
 
     Dependency(String group, String name, String version, String...exc) {
         this.group = group
